src/plugins/kibana_react/public/page_template/with_solution_nav.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { ComponentType, useState } from 'react';
+import classNames from 'classnames';
+import { useIsWithinBreakpoints } from '@elastic/eui';
+import { EuiPageSideBarProps } from '@elastic/eui/src/components/page/page_side_bar';
+import {
+  KibanaPageTemplateSolutionNav,
+  KibanaPageTemplateSolutionNavProps,
+} from '../page_template/solution_nav';
+import { KibanaPageTemplateProps } from '../page_template';
+
+type SolutionNavProps = KibanaPageTemplateProps & {
+  solutionNav: KibanaPageTemplateSolutionNavProps;
+};
+
+export const withSolutionNav = (WrappedComponent: ComponentType<KibanaPageTemplateProps>) => {
+  const WithSolutionNav = (props: SolutionNavProps) => {
+    const isMediumBreakpoint = useIsWithinBreakpoints(['m']);
+    const isLargerBreakpoint = useIsWithinBreakpoints(['l', 'xl']);
+    const [isSideNavOpenOnDesktop, setisSideNavOpenOnDesktop] = useState(
+      !JSON.parse(String(localStorage.getItem('solutionNavIsCollapsed')))
+    );
+    const { solutionNav, children, isEmptyState, template } = props;
+    const toggleOpenOnDesktop = () => {
+      setisSideNavOpenOnDesktop(!isSideNavOpenOnDesktop);
+      // Have to store it as the opposite of the default we want
+      localStorage.setItem('solutionNavIsCollapsed', JSON.stringify(isSideNavOpenOnDesktop));
```

Comment:
nit: make this a constant, since we use it in a few places.

We should consider a hook for local/session storage.